### PR TITLE
Correct ff/no-ff functionality for Dolt_Merge() and add tests for DOLT_MERGE when autocommit is off

### PR DESCRIPTION
**Pull request by VinaiRachakonda**

### CODE REVIEW

File: integration-tests/bats/sql-merge.bats [hunk 1]
```diff
@@ -88,8 +88,84 @@ SELECT DOLT_MERGE('feature-branch');
 -- commit;
```
Comment (timsehn) [inline integration-tests/bats/sql-merge.bats:88]: Kill this comment.

**Review by timsehn — APPROVED** [review]

**Comment by timsehn** [comment]
So if autocommit is on for Prado and you say this fixes his issue. I think you have to change this test:

https://github.com/dolthub/dolt/blob/5c287c79a09cce484628deea5f72901d06b56515/integration-tests/bats/sql-merge.bats#L65

to verify using dolt status like he did.

File: integration-tests/bats/sql-merge.bats [hunk 2]
```diff
@@ -71,25 +71,104 @@ SELECT DOLT_COMMIT('-a', '-m', 'this is a ff');
 SELECT DOLT_CHECKOUT('main');
 SELECT DOLT_MERGE('feature-branch');
 SELECT COUNT(*) > 0 FROM test WHERE pk=3;
+
+SELECT COUNT(*) from dolt_status;
```
Comment (timsehn) [inline integration-tests/bats/sql-merge.bats:75]: Move this into a `dolt sql -q`  below the other test. Then assert there's nothing in dolt_status

Comment (timsehn) [reply integration-tests/bats/sql-merge.bats:75]: You can also do something clever with something like 
```
test-nickname $ dolt sql -r csv -q "select hash from dolt_branches where branch='main'"
hash
4mgglt0kr56l9c0joff7dt2nbjt815bh
```
do another select from dolt_branches for feature-branch , and then assert that the two hashes are the same.

File: integration-tests/bats/sql-merge.bats [hunk 3]
```diff
@@ -74,22 +74,109 @@ SELECT COUNT(*) > 0 FROM test WHERE pk=3;
...
+    [ "${lines[4]}" = "3" ]
+}
+
+@test "sql-merge: End to End Conflict Resolution with autocommit off." {
+    run dolt sql << SQL
+    CREATE TABLE test2 (pk int primary key, val int);
+    INSERT INTO test2 VALUES (0, 0);
+    SET autocommit = 0;
+    SELECT DOLT_COMMIT('-a', '-m', 'Step 1');
+		SELECT DOLT_CHECKOUT('-b', 'feature-branch');
```
Comment (timsehn) [inline integration-tests/bats/sql-merge.bats:139]: weird indenting here

**Review by timsehn — APPROVED** [review]
Fix the indenting and the ship this. Nice work.

File: integration-tests/bats/sql-merge.bats [hunk 4]
```diff
@@ -74,22 +74,109 @@ SELECT COUNT(*) > 0 FROM test WHERE pk=3;
...
 SELECT DOLT_CHECKOUT('-b', 'feature-branch');
 INSERT INTO test VALUES (3);
 SELECT DOLT_COMMIT('-a', '-m', 'this is a ff');
 SELECT DOLT_CHECKOUT('main');
 SELECT DOLT_MERGE('feature-branch');
--- commit;
 SELECT DOLT_CHECKOUT('-b', 'new-branch');
 SQL
-    skip "That commented out commit; is necessary now. It should not be."
     [ $status -eq 0 ]
```
Comment (zachmu) [inline integration-tests/bats/sql-merge.bats:102]: For these tests, if the only assertion  you're making is status == 0, then you can just remove 'run', it's equivalent (and gives better error info if it fails)

File: integration-tests/bats/sql-merge.bats [hunk 5]
```diff
@@ -74,22 +74,109 @@ SELECT COUNT(*) > 0 FROM test WHERE pk=3;
...
+SQL
+    [ $status -eq 0 ]
+
+    run dolt sql -r csv -q "select * from test2 order by pk"
+    [ "${#lines[@]}" -eq 3 ]
+    [ "${lines[1]}" = "0,1001" ]
+    [ "${lines[2]}" = "1,1" ]
+}
+
+@test "sql-merge: DOLT_MERGE works with autocommit off." {
```
Comment (zachmu) [inline integration-tests/bats/sql-merge.bats:158]: Should have a skipped test for a merge with conflicts when autocommit = on

**Review by zachmu — APPROVED** [review]
Nice work, no major comments

File: go/libraries/doltcore/sqle/enginetest/dolt_queries.go [hunk 6]
```diff
@@ -88,3 +90,365 @@ var DoltScripts = []enginetest.ScriptTest{
...
+			"SELECT DOLT_COMMIT('-a', '-m', 'this is a ff');",
+			"SELECT DOLT_CHECKOUT('main');",
+		},
+		Assertions: []enginetest.ScriptTestAssertion{
+			{
+				Query:    "SELECT DOLT_MERGE('feature-branch', '--squash')",
+				Expected: []sql.Row{{1}},
+			},
+			{
+				Query:       "SELECT DOLT_CHECKOUT('-b', 'other')",
```
[the remark this answers is not in the document]
Comment (VinaiRachakonda) [reply go/libraries/doltcore/sqle/enginetest/dolt_queries.go:308]: https://github.com/dolthub/dolt/issues/2723

File: go/libraries/doltcore/sqle/enginetest/dolt_queries.go [hunk 7]
```diff
@@ -88,3 +90,365 @@ var DoltScripts = []enginetest.ScriptTest{
...
+		},
+	},
+	{
+		Name: "DOLT_MERGE with dolt_branches table",
+		SetUpScript: []string{
+			"CREATE TABLE test (pk int primary key)",
+			"INSERT INTO test VALUES (0),(1),(2);",
+			"SELECT DOLT_COMMIT('-a', '-m', 'Step 1');",
+			"SELECT DOLT_CHECKOUT('-b', 'feature-branch')",
+			"UPDATE dolt_branches SET hash = HASHOF('main^') WHERE name='feature-branch'",
```
[the remark this answers is not in the document]
Comment (VinaiRachakonda) [reply go/libraries/doltcore/sqle/enginetest/dolt_queries.go:324]: Was referencing this issue: https://github.com/dolthub/dolt/issues/2705

**Pull request merged**